template.distillery/PROJECT_NAME_language.eliom

```diff
@@ -0,0 +1,128 @@
+(* This file was generated by Ocsigen-start.
+   Feel free to use it, modify it, and redistribute it as you wish. *)
+
+let%shared language_of_string s =
+ let s = if String.length s > 2 then String.sub s 0 2 else s in
+ match s with
+ | "fr" -> %%%MODULE_NAME%%%_i18n.Fr
+ | "en" -> %%%MODULE_NAME%%%_i18n.Us
+ | _ -> raise Not_found
+
+let%shared string_of_language = function
+ | %%%MODULE_NAME%%%_i18n.Fr -> "fr-FR"
+ | %%%MODULE_NAME%%%_i18n.Us -> "en-US"
+
+let%server best_matched_language () =
+  (* lang contains a list of (language_as_string, quality_value) *)
+  let lang = Eliom_request_info.get_accept_language () in
+  (* If no quality is given, we suppose it's 1 *)
+  let lang = List.map (fun (s, quality_value) ->
+    let quality_value = match quality_value with
+    | None -> 1.
+    | Some quality_value -> quality_value
+    in
+    (s, quality_value))
+    lang
+  in
+  (* Increasingly sort based on the quality *)
+  let lang = List.sort (fun (_, q1) (_, q2) -> compare q2 q1) lang in
+  let lang = List.map (fun (language, _) -> language_of_string language) lang in
+  (* The first language of the list is returned. If the list is empty, the
+     default language is returned.
+   *)
+  Lwt.return
+    (match lang with
+     | [] -> (%%%MODULE_NAME%%%_i18n.default_language)
+     | language :: _ -> language)
+
+let%shared get_language () =
+  %%%MODULE_NAME%%%_i18n.get_lang ()
+
+let%shared set_language lang =
+  %%%MODULE_NAME%%%_i18n.set_lang lang
+
+let%server update_language lang =
+  let language  = string_of_language lang in
+  let myid_o    = Os_current_user.Opt.get_current_userid () in
+  (* Update the server and client values *)
+  %%%MODULE_NAME%%%_i18n.set_lang lang;
+  ignore [%client (%%%MODULE_NAME%%%_i18n.set_lang ~%lang : unit)];
+  (* Update in the database if a user is connected *)
+  match myid_o with
+  | None -> Lwt.return ()
+  | Some userid -> Os_user.update_language ~userid ~language
+
+let%server _ =
+  Os_session.on_start_process
+    (fun () ->
+       (* Guess a default language. *)
+       let%lwt lang = best_matched_language () in
+       ignore (update_language lang);
+       Lwt.return ());
+  Os_session.on_start_connected_process
+    (fun userid ->
+      (* Set language according to user preferences. *)
+      let%lwt lang = Os_user.get_language userid in
+      let language = match lang with
+      | Some language -> (language_of_string language)
+      | None ->
+        (ignore (Os_user.update_language
+          ~userid
+          ~language:(
+            string_of_language %%%MODULE_NAME%%%_i18n.default_language));
+        %%%MODULE_NAME%%%_i18n.default_language)
+      in
+      set_language language;
+      ignore ([%client (set_language ~%language : unit)]);
+      Lwt.return ())
+
+let%shared _ =
+  Os_i18n.passwords_do_not_match :=
+    [%i18n S.passwords_do_not_match];
```

Comment:
I think the error comes from here when the translation is done (variables are not defined...).